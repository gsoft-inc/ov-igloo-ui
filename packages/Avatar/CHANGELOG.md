# @igloo-ui/avatar

## 0.3.3

### Patch Changes

- 81723d0: Updated Hopper depedency

## 0.3.2

### Patch Changes

- 1613653: Updated Hopper Tokens Fonts package

## 0.3.1

### Patch Changes

- 6ea531f: Updated dependency versions for react-aria packages, luxon and hopper

## 0.3.0

### Minor Changes

- 4846c59: Wrapped Igloo CSS in @layer to isolate it.

## 0.2.4

### Patch Changes

- cfb5619: Updated hopper token versions and button versions.

## 0.2.3

### Patch Changes

- 7d36815: Updated hopper token versions and igloo icon versions.

## 0.2.2

### Patch Changes

- a2df55d: Updated to latest Hopper Token dependency

## 0.2.1

### Patch Changes

- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.

## 0.2.0

### Minor Changes

- 5402bcc: Created a "brand" toggle in the Toolbar and applied the new Workleap brand to some components. Also deprecated the Tag's appearance values and added new ones.

## 0.1.1

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console

## 0.1.0

### Minor Changes

- 2a6d8e5: Extracted the Color and Avatar components from the Tag component

  ## BREAKING CHANGE

  Restructuring HTML in the tag component as well as renaming onClose to OnRemove since it made more sense. onRemove also gains an id as a parameter.
