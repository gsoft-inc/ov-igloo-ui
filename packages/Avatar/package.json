{
  "name": "@igloo-ui/avatar",
  "version": "0.3.3",
  "main": "dist/Avatar.js",
  "module": "dist/Avatar.js",
  "types": "dist/Avatar.d.ts",
  "styles": "dist/avatar.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Vicky Comeau vicky.comeau@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^2.1.0",
    "@hopper-ui/tokens": "^4.2.0",
    "classnames": "^2.3.2"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
