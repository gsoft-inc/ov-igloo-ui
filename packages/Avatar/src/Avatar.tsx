import * as React from "react";

import cx from "classnames";

import "./avatar.scss";

export type Size = "xsmall" | "small" | "medium" | "large" | "xlarge";

export interface AvatarProps extends React.ComponentProps<"img"> {
    /** The alternate text for the image */
    alt?: string;
    /** Add a specific class to the avatar */
    className?: string;
    /** Add a data-test tag for automated tests */
    dataTest?: string;
    /** Specifies the size of the avatar */
    size?: Size;
    /** Specifies the image url to show */
    src: string;
}

const Avatar: React.FunctionComponent<AvatarProps> = ({
    alt = "Avatar",
    className,
    dataTest,
    size = "medium",
    src,
    ...rest
}) => {
    const classes = cx("ids-avatar", `ids-avatar--${size}`, className);

    const checkImgSrc = (): string => {
        const validBase64Regex =
      /^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$/;

        let image = src;
        if (validBase64Regex.exec(image) !== null) {
            image = `data:image/jpeg;base64,${image}`;
        }

        return image;
    };

    return (
        <div className={classes}>
            <img
                alt={alt}
                className={cx("ids-avatar__image", `ids-avatar__image--${size}`)}
                data-test={dataTest}
                src={checkImgSrc()}
                {...rest}
            />
        </div>
    );
};

export default Avatar;
