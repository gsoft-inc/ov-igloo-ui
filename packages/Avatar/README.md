# Avatar

An image icon used to identify a user.

<Example />

<ReferenceLinks />

## Installation

To install `@igloo-ui/avatar` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/avatar
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/avatar
```

## Usage

Then to use the component in your code just import it!

```jsx
import Avatar from '@igloo-ui/avatar';

<Avatar
  size="medium"
  src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQEB3tCC4oJsa4ZZsiEDIhWi06EyN3iHYMoOg&usqp=CAU"
/>;
```
