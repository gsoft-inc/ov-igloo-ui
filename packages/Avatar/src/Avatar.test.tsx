/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import { toHaveNoViolations } from 'jest-axe';

import axe from '../../../axe-helper.js';

import Avatar, { Size } from './Avatar';

const setup = (props = {}) => {
  return render(
    <Avatar
      dataTest="ids-avatar"
      src="https://cdn-icons-png.flaticon.com/512/168/168724.png"
      {...props}
    />
  );
};

expect.extend(toHaveNoViolations);

describe('Avatar', () => {
  const expectToBeOfSize = (component: RenderResult, size: Size) => {
    const avatar = component.container.firstChild;
    const expectedClass = 'ids-avatar--' + size;
    expect(avatar).toHaveClass(expectedClass);
  };

  test('It should render without errors', () => {
    setup();
    const wrapper = screen.getByTestId('ids-avatar');
    expect(wrapper).toBeInTheDocument();
  });

  test('It should render a snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  test('It should render a xsmall avatar', () => {
    const avatar = setup({ size: 'xsmall' });

    expectToBeOfSize(avatar, 'xsmall');
  });

  test('It should render a small avatar', () => {
    const avatar = setup({ size: 'small' });

    expectToBeOfSize(avatar, 'small');
  });

  test('It should render a medium avatar', () => {
    const avatar = setup({ size: 'medium' });

    expectToBeOfSize(avatar, 'medium');
  });

  test('It should render a large avatar', () => {
    const avatar = setup({ size: 'large' });

    expectToBeOfSize(avatar, 'large');
  });

  test('It should render a xlarge avatar', () => {
    const avatar = setup({ size: 'xlarge' });

    expectToBeOfSize(avatar, 'xlarge');
  });

  test('It should have no accessibility violations', async () => {
    const { container } = setup();
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
