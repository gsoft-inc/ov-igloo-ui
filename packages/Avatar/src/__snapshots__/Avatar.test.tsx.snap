// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar It should render a snapshot 1`] = `
<DocumentFragment>
  <div
    class="ids-avatar ids-avatar--medium"
  >
    <img
      alt="Avatar"
      class="ids-avatar__image ids-avatar__image--medium"
      data-test="ids-avatar"
      src="https://cdn-icons-png.flaticon.com/512/168/168724.png"
    />
  </div>
</DocumentFragment>
`;
